File: node/server-single.js
var http = require("http"),
	fs = require("fs"),
	urlparse = require("url").parse,
	uuid = require("node-uuid")




http.createServer(function(req,res){
	var parsed = urlparse(req.url,true);
	var path = parsed.pathname;

	var components = path.split("/").filter(function(x){return x!=""});
	
	if (components.length < 2 ||
		components[0] != "grounding" ||
		components[1].length != 5
		) {
		res.writeHead("402");
		res.end("Bad request.");
		return;
	}

	var fileid = uuid.v4()

	var dest = fs.createWriteStream(__dirname+"/../data/"+fileid+".xml")
	req.pipe(dest);
	req.on("end",function(){
		res.writeHead(200);
		res.end("Message received");
	});

}).listen(9000);
console.log("Listening on port 9000");

File: go/server-single.go
package main

import (
	"fmt"
	"net/http"
	"runtime"
	"log"
	"strings"
	"go-uuid/uuid"
	"os"
	"io"
)

type GroundingServer struct {}

func PipeRequest(r *http.Request, path string) {
	if(r.Body != nil) {
		f,err := os.Create(path)
		if err == nil {
			io.Copy(f, r.Body)
			f.Close()
		} else {
			log.Fatal(err)
		}
	}
}

func (gs GroundingServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	//fmt.Println(r.URL)

	path := r.URL.Path
	var compolents []string
	//var params map[string]string
	//params := r.URL.Query()
	strs:=strings.Split(path,"/")

	for i := range strs {
		s := strs[i]
		if s != "" {
			compolents = append(compolents,s)
		}
	}
	//fmt.Println(compolents)
	//fmt.Println(params)

	fileid := uuid.NewRandom().String()

	//fmt.Println(fileid)

	if(compolents[0] != "grounding" || len(compolents[1]) != 5 || len(compolents) > 2) {
		
		w.WriteHeader(402);
		w.Write([]byte("Bad request"));
		return
	}

	filepath := "../data/" + fileid + ".xml"
	PipeRequest(r, filepath);
	w.Write([]byte("Message received."));
}

func main() {
	runtime.GOMAXPROCS(1)

	var gs GroundingServer
    log.Fatal(http.ListenAndServe("172.7.1.16:9001",gs))
    fmt.Println("Listening on port: 9001");
}
